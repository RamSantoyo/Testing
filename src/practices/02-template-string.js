
 export function getSaludo(nombre) {
    return 'Hola ' + nombre;
}

export function getSuma(num1, num2) {
    return num1 + num2;
}