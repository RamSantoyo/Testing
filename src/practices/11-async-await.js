

// const getImagenPromesa = () => new Promise( resolve => resolve('https://ajskdhaskjdhajs.com') )
// getImagenPromesa().then( console.log );

export const getImagen = async (name) => {
    try {
        const resp = await fetch(`https://pokeapi.co/api/v2/pokemon/${name}`);
        const data = await resp.json();
        const { front_default } = data.sprites;
        return front_default;
    } catch (error) {
        //console.error(error);
        return undefined;
    }
};



