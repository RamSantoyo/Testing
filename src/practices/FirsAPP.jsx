import React from 'react';
import PropTypes from 'prop-types';

export const FirstApp = ({ title = "Default Title", subTitle = "Default Subtitle", name = "Default Name" }) => {

  return (
    <>
      <h1>{title}</h1>
      <p>{subTitle}</p>
      <p>{name}</p>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired
}
