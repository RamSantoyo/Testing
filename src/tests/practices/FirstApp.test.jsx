import { render } from '@testing-library/react';
import {FirstApp} from '../../practices/FirsAPP'; 

describe('Renderizar el componente <FirstApp />', () => {
  test('Render component', () => {
    const title = 'Hola, soy Goku';
    const subTitle = 'Soy un subtitulo';
    const name = 'Goku';
    render(<FirstApp title={title} subTitle={subTitle} name={name} />);
  });

  test('Render match with snapshot', () => {
    const title = 'Hola, soy Goku';
    const subTitle = 'Soy un subtitulo';
    const name = 'Goku';
    const {container} = render(<FirstApp title={title} subTitle={subTitle} name={name} />);
    //expect(container).toMatchSnapshot(); comparar el snapshot
  });

  test('Default props render', () => {
    const subTitle = 'Soy un subtitulo';
    const name = 'Goku';
    const {container} = render(<FirstApp subTitle={subTitle} name={name} />);
    expect(container).toMatchSnapshot();
  })

  test('Watching H1 title', () => {
    const title = 'Hola, soy Ramsess';
    const subTitle = 'Soy un subtitulo';
    const name = 'Ramsess';    
    const {container, getByText} = render(<FirstApp title={title} subTitle={subTitle} name={name} />);
    expect(getByText(title)).toBeTruthy();// si existe el texto

    const h1 = container.querySelector('h1');
    expect(h1.innerHTML).toBe(title);

    const titles = container.querySelectorAll('h1');
    expect(titles.length).toBe(1);
  })
 
});


//npm run file -- -u  (actualiza el snapshot)