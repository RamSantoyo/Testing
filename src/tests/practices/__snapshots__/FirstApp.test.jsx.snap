// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderizar el componente <FirstApp /> Default props render 1`] = `
<div>
  <h1>
    Default Title
  </h1>
  <p>
    Soy un subtitulo
  </p>
  <p>
    Goku
  </p>
</div>
`;
